src/libcore/convert.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright 2014 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Traits for conversions between types.
+//!
+//! The traits in this module provide a general way to talk about
+//! conversions from one type to another. They follow the standard
+//! Rust conventions of `as`/`to`/`into`/`from`.
+
+#![unstable(feature = "convert",
+            reason = "recently added, experimental traits")]
+
+use option::Option;
+use result::Result;
+use marker::Sized;
+
+/// A cheap, reference-to-reference conversion.
+pub trait As<T: ?Sized> {
+    /// Perform the conversion.
+    fn convert_as(&self) -> &T;
```

Comment:
I do quite like your suggestions on the RFC of `as_ref`, `as_mut`, `to`, and `into`. The only one that's "wrong" is `as_ref` (e.g. it should be called `as`) but it seems worth it!